lib/WasmReader/WasmBinaryReader.cpp

```diff
@@ -177,12 +177,12 @@ WasmBinaryReader::ProcessSection(SectionCode sectionId, bool isEntry /*= true*/)
         break; // This section is not used by bytecode generator, stay in decoder
     }
     case bSectDataSegments:
-        // TODO: Populate Data entry info
-        ReadConst<UINT32>();  // dest addr in module memory
-        ReadConst<UINT32>();  // source offset (?)
-        ReadConst<UINT32>();  // size
-        ReadConst<UINT8>();  // init
-        m_moduleState.count++;
+        m_moduleInfo->AllocateDataSegs(m_moduleState.size);
```

Comment:
Good catch. Still used to the prior version where size was initialized outside the switch.